code/game/machinery/_machinery.dm

```diff
@@ -256,32 +265,25 @@ Class Procs:
 		user.visible_message("<span class='danger'>[user.name] smashes against \the [src.name] with its paws.</span>", null, null, COMBAT_MESSAGE_RANGE)
 		take_damage(4, BRUTE, "melee", 1)
 
+/obj/machinery/attack_robot(mob/user)
+	if(!(interaction_flags_machine & INTERACT_MACHINE_ALLOW_SILICON) && !IsAdminGhost(user))
+		return FALSE
+	var/mob/living/silicon/robot/R = user
+	if(in_view_range(R, src)  && !R.low_power_mode)
```

Comment:
Generally dislike binding this to view range, having remote interaction range be separate you can unify ai intellicard and cyborg interaction with something common.